portfolio/src/main/webapp/script.js

```diff
@@ -15,14 +15,56 @@
 /**
  * Adds a random greeting to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function randomizeImage() {
+  // The images directory contains 13 images, so generate a random index between
+  // 1 and 13.
+  const imageIndex = Math.floor(Math.random() * 24) + 1;
+  const imgUrl = 'images/jfew-' + imageIndex + '.jpg';
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  const imgElement = document.createElement('img');
+  imgElement.src = imgUrl;
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  const imageContainer = document.getElementById('random-image-container');
+  // Remove the previous image.
+  imageContainer.innerHTML = '';
+  imageContainer.appendChild(imgElement);
+}
+
+function openIntroModal() {
+  const modal = document.getElementById("myIntroModal");
+  modal.style.display = "block";
+}
+
+function openPictureModal() {
+  const modal = document.getElementById("myPictureModal");
+  modal.style.display = "block";
+}
+
+function openConnectModal() {
+  const modal = document.getElementById("myConnectModal");
+  modal.style.display = "block";
+}
+
+function closeIntroModal(){
+  const modal = document.getElementById("myIntroModal");
+  modal.style.display = "none";
+}
+
+function closePictureModal(){
+  const modal = document.getElementById("myPictureModal");
+  modal.style.display = "none";
+}
+
+function closeConnectModal(){
+  const modal = document.getElementById("myConnectModal");
+  modal.style.display = "none";
+}
+
+window.onclick = function(event) {
```

Comment:
this is one way to do it, but it means that for every click for every window, you'll run this code, which seems a bit unnecessary. 

there's a different approach here https://stackoverflow.com/a/54441029 that looks promising and reduces code duplication if you want a challenge